pkg/controller/metrics/metrics.go

```diff
@@ -819,6 +818,54 @@ func readMetricsConfig() (*metricsconfig.MetricsConfig, error) {
 	return config, nil
 }
 
+// GetOptionalClusterTypeLabels reads the MetricsWithClusterTypeLabels from the metrics config and returns the same as a
+// map, as well as a slice of metric labels. The latter is helpful for defining the relevant metrics
+func GetOptionalClusterTypeLabels(mConfig *metricsconfig.MetricsConfig) (map[string]string, []string) {
```

Comment:
As discussed in slack, IMO we should force the user to specify `hive.openshift.io/cluster-type` explicitly if they want it. To accommodate a smoother transition, we could default to that if the config isn't used:

```go
if mConfig.MetricsWithClusterTypeLabels == nil {
    return map[string]string{"cluster_type": "hive.openshift.io/cluster-type"}, []string{"cluster_type"}
}
// ... else carry on with what you've got below
```